### PR TITLE
Add Makefile and Travis CI build

### PR DESCRIPTION
**Pull request by songy23**
Follow up of https://github.com/open-telemetry/opentelemetry-specification/pull/199#issuecomment-516776682.

Need repo admins to set up Travis after this commit.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,10 @@
 .settings
 bin
 
+# Go module
+go.mod
```
Comment (bogdandrutu) [inline .gitignore]: for my curiosity, why this? We may need to update this.

Comment (songy23) [reply .gitignore]: On a second thought I agree it's better to have the modules. Also added a `tools.go` to follow the best practice on having a consistent version on tools in https://github.com/open-telemetry/opentelemetry-specification/pull/200/commits/7b79cea19564f270599b29a8150f3f6948695335.

File: .gitignore [hunk 2]
```diff
@@ -20,4 +20,4 @@ bin
 \#*\#
 
 # Vim
-.swp
\ No newline at end of file
+.swp
```
Comment (bogdandrutu) [inline .gitignore]: Exclude this file changes?

Comment (songy23) [reply .gitignore]: Done - seems my editor did it.

**Pull request merged**